### PR TITLE
Replace ApiVersion enum with a looser string union type

### PR DESCRIPTION
**Pull request by peter-hamilton**
## Description

Fixes (issue #1895)

This PR converts the ApiVersion (used in `koa-shopify-webhooks` and `koa-shopify-graphql-proxy`) from an `enum` to a union that allows for any string while still providing suggestions. This change allows developers to use new versions without waiting for the enum to be updated in the package.

Breaking: Any code that used the previous enum type values  will have to be changed to string values:

`ApiVersion.Unstable` ==> `'unstable'`, `ApiVersion.April20` ==> `'2020-04'`

## Type of change

- [x] `koa-shopify-webhooks` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] `koa-shopify-graphql-proxy` Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/koa-shopify-graphql-proxy/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 ### Changed
 
 - Updated build tooling, types are now compiled with TypeScript 4.3. [[#1997](https://github.com/Shopify/quilt/pull/1997)]
+- Replace ApiVersion enum with a looser string union type [[#2001](https://github.com/Shopify/quilt/pull/2001)]
```
[the remark this answers is not in the document]
Comment (atesgoral) [reply packages/koa-shopify-graphql-proxy/CHANGELOG.md:13]: ... or something like that

File: packages/koa-shopify-webhooks/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 ### Changed
 
 - Updated build tooling, types are now compiled with TypeScript 4.3. [[#1997](https://github.com/Shopify/quilt/pull/1997)]
+- Replace ApiVersion enum with a looser string union type [[#2001](https://github.com/Shopify/quilt/pull/2001)]
```
Comment (atesgoral) [inline packages/koa-shopify-webhooks/CHANGELOG.md:13]:
```suggestion
- Replace ApiVersion enum with a looser string union type and update supported versions [[#2001](https://github.com/Shopify/quilt/pull/2001)]
```

File: packages/koa-shopify-graphql-proxy/CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (BPScott) [reply packages/koa-shopify-graphql-proxy/CHANGELOG.md:13]: This is a breaking change. Can you put it under a `### Breaking Change` heading? Same for the other changelog

[the remark this answers is not in the document]
Comment (peter-hamilton) [reply packages/koa-shopify-graphql-proxy/CHANGELOG.md]: Applied.

File: packages/koa-shopify-webhooks/CHANGELOG.md (hunk 2 above)
Comment (peter-hamilton) [reply packages/koa-shopify-webhooks/CHANGELOG.md]: Applied.

**Comment by atesgoral** [comment]
proxy.test.ts has an unused `ApiVersion` import

**Comment by BPScott** [comment]
> proxy.test.ts has an unused ApiVersion import

Making type-checks run over test files (so a`yarn type-check` would catch things like these unused imports) and then go fixup whatever it grumbles about is on my list but that'd actually be a pretty good getting to know things task for @peter-hamilton if he's interested

**Review by atesgoral — APPROVED** [review]

**Comment by Resousse** [comment]
great!

**Pull request merged**